### PR TITLE
Upgrade to Gitlab CE 14.9.0

### PR DESCRIPTION
**Pull request by omueller**
https://about.gitlab.com/releases/2022/03/22/gitlab-14-9-released/ 

Changes after review

* gitlab: upgrade CE to v14.9.0 
* gitaly: upgrade to v14.9.0
* gitlab-pages: upgrade to v1.56.0
* gitlab-shell: upgrade to v13.24.0

Details

* before review: gitlab-shell: upgrade to v13.24.1,
  after review: gitlab-shell: upgrade to v13.24.0, since upstream also "just" uses this version (intended to avoid side effects)
* ubuntu: upgrade to focal-20220316 (used by previous versions, therefore no real change)

### CODE REVIEW

**Comment by omueller** [comment]
This was a first attempt to a bigger release.  It fails on something strange, not sure how to fix/debug this.  Maybe update of GOLANG_VERSION and/or GITLAB_SHELL_VERSION ?    

```
go: downloading github.com/jmespath/go-jmespath v0.4.0
gofmt -w -s ./vendor/cloud.google.com/go/profiler/mutex.go ./vendor/cloud.google.com/go/profiler/heap.go (...) check/command/command_test.go ./cmd/check/main.go ./cmd/check/command/command.go ./cmd/check/command/command_test.go | awk '{ print } END { if (NR > 0) { print "Please run make fmt"; exit 1 } }'
make: execvp: /bin/sh: Argument list too long
make: *** [Makefile:18: verify_golang] Error 127
The command '/bin/sh -c bash ${GITLAB_BUILD_DIR}/install.sh' returned a non-zero code: 2

```

**Comment by kkimurak** [comment]
I have confirmed that build passed after deleting format and verification step in [`assets/build/install.sh`](https://github.com/sameersbn/docker-gitlab/blob/739480e81d20b3fc5f3f5a721fde1a4a4041de44/assets/build/install.sh#L113-L114). That was a workaround I have introduced to build gitlab-shell v13.21.0 for gitlab v14.3.0 release (see reports in #2425 , #2435). It seems to be [fixed on upstream a month ago](https://gitlab.com/gitlab-org/gitlab-shell/-/merge_requests/565) so we don't need them anymore.

Could you try this patch ? :
```diff
diff --git a/assets/build/install.sh b/assets/build/install.sh
index 4161b52d..a41ead05 100755
--- a/assets/build/install.sh
+++ b/assets/build/install.sh
@@ -110,8 +110,7 @@ echo "Compiling gitlab-shell golang executables..."
 exec_as_git bundle config set --local deployment 'true'
 exec_as_git bundle config set --local with 'development test'
 exec_as_git bundle install -j"$(nproc)"
-exec_as_git "${GOROOT}"/bin/go mod vendor
-exec_as_git "PATH=$PATH" make fmt verify setup
+exec_as_git "PATH=$PATH" make verify setup
 
 # remove unused repositories directory created by gitlab-shell install
 rm -rf ${GITLAB_HOME}/repositories
```

**Comment by omueller** [comment]
Thanks for your comment @kkimurak, fix added, let's see how it looks like after the test build.

**Comment by omueller** [comment]
Looking good :)

**Comment by sachilles** [comment]
@omueller and @kkimurak Really good work. But now we should hurry up, because version 14.9.1 is on the way.

Nevertheless, I have seen that GitLab 14.9.0 includes the following versions for its components:

- GITALY_SERVER_VERSION 14.9.0
- **GITLAB_SHELL_VERSION 13.24.0 (and not 13.24.1, GitLab 14.9.1 even uses 13.24.0)**
- GITLAB_PAGES_VERSION 1.56.0

If necessary, go can be updated to version 1.17.8 or 1.18.0. (Otherwise it could be wise to wait for version 1.18.1.)

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+ARG VERSION=14.9.0
 
 ENV GITLAB_VERSION=${VERSION} \
     RUBY_VERSION=2.7.5 \
     RUBY_SOURCE_SHA256SUM="2755b900a21235b443bb16dadd9032f784d4a88f143d852bc5d154f22b8781f1" \
     GOLANG_VERSION=1.17.7 \
-    GITLAB_SHELL_VERSION=13.23.2 \
-    GITLAB_PAGES_VERSION=1.54.0 \
-    GITALY_SERVER_VERSION=14.8.2 \
+    GITLAB_SHELL_VERSION=13.24.1 \
```
Comment (sachilles) [inline Dockerfile:9]: The version seems to be inkonsistent with https://gitlab.com/gitlab-org/gitlab-foss/-/blob/v14.9.0/GITLAB_SHELL_VERSION .

File: Changelog.md [hunk 2]
```diff
@@ -2,6 +2,17 @@
 
 This file only reflects the changes that are made in this image. Please refer to the upstream GitLab [CHANGELOG](https://gitlab.com/gitlab-org/gitlab-foss/blob/master/CHANGELOG.md) for the list of changes in GitLab.
 
+**14.9.0**
+- gitlab: upgrade CE to v14.9.0 - https://about.gitlab.com/releases/2022/03/22/gitlab-14-9-released/
+- gitaly: upgrade to v14.9.0
+- gitlab-pages: upgrade to v1.56.0
+- gitlab-shell: upgrade to v13.24.1
```
Comment (sachilles) [inline Changelog.md:9]: The version seems to be inkonsistent with https://gitlab.com/gitlab-org/gitlab-foss/-/blob/v14.9.0/GITLAB_SHELL_VERSION .

File: Dockerfile [hunk 3]
```diff
@@ -1,14 +1,14 @@
-FROM ubuntu:focal-20220113
+FROM ubuntu:focal-20220316
 
-ARG VERSION=14.8.2
+ARG VERSION=14.9.0
 
 ENV GITLAB_VERSION=${VERSION} \
     RUBY_VERSION=2.7.5 \
     RUBY_SOURCE_SHA256SUM="2755b900a21235b443bb16dadd9032f784d4a88f143d852bc5d154f22b8781f1" \
     GOLANG_VERSION=1.17.7 \
```
Comment (sachilles) [inline Dockerfile:8]: Could you be so kind and upgrade to version 1.17.8?

**Review by sachilles — COMMENTED** [review]
I could not find any notes for the changes between version 13.24.0 and 13.24.1 of the GitLab-Shell. Should we just update to 13.24.0 to be on the safe side?

File: Dockerfile (hunk 3 above)
Comment (sachilles) [reply Dockerfile]: Resolved automatically because of other merges.

**Comment by sachilles** [comment]
It is possible that I come across as mouthy, but that is not what I have in mind.

I thank you very much for the excellent draft.

**Review by sachilles — APPROVED** [review]
LGTM

File: Changelog.md (hunk 2 above)
Comment (omueller) [reply Changelog.md]: thanks a lot for the catch and PR update @sachilles, noted for next time!

**Pull request merged**